src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -31,4 +38,28 @@ public void arcadeDrive(double drive, double rotate)
 
         tankDrive(left, right);
     }
+
+    // not a good solution
+    public void drive(double left, double right, double drive, double rotate)
+    {
+        if (_driveMode == DriveMode.Arcade)
+        {
+            arcadeDrive(drive, rotate);
+        }
+        
+        else
+        {
+            tankDrive(left, right);
+        }
+    }
+
+    public void changePreferredDriveMode()
+    {
+        _driveMode = (_driveMode == DriveMode.Tank) ? DriveMode.Arcade : DriveMode.Tank;
+    }
+
+    public DriveMode getPreferredDriveMode()
+    {
+        return _driveMode;
+    }
```

Comment:
Does it make sense to monitor the preferred drive mode in the subsystem? I would think it'd be more appropriate to do this in the command.

The subsystems provide an interface to the hardware, but the commands dictate what the subsystem does and when.

Reply:
I put the preferred driving mode in the subsystem because I figured it'd be easier to change it with an instantcommand on button press. I could move the method to the command and run the method with an instantcommand instead.

Reply:
I'll fix the indents while I'm at it